lib/state_of_the_nation.rb

```diff
@@ -33,6 +33,24 @@ def until(finish_key)
         }
       end
 
+      private def round_if_should(time)
```

Comment:
This was an interesting Ruby thing I learned today.

What we have here looks like nested methods but Ruby doesn't support nested methods so what is actually happening is when `considered_active` is called it creates the `from` and `until` methods on the same instance it was called on and returns `self`.

If we look at its usage `considered_active.from(:started_at).until(:cancelled_at)` is called when the class is loaded only `considered_active` is present, calling that then creates the new `from` and `until` instance methods and allows for the chaining we see because `self` is returned from `considered_active` and `from`.

Why these methods need to be moved into `considered_active` is for the same reason `from` and `until` are defined in there, so they're defined only when `considered_active` is used. We could create all of these instance methods in the module but that's the interface now 🤷‍♂ 

I'm not sure why this worked in the previous version of ruby but it makes sense why it does now.